Update `CONTRIBUTING.md` > `Contributing to` > `The compiler itself` - broken link, outdated steps and stated LLVM version

[`CONTRIBUTING.md` > `Contributing to` > `The compiler itself`](https://github.com/crystal-lang/crystal/blame/master/CONTRIBUTING.md#L113) currently:

- Refers to steps in a broken link to a [non-existent .travis.yml](https://github.com/crystal-lang/crystal/blame/master/CONTRIBUTING.md#L120)
- Refers to LLVM 3.6 when current versions stated in [src/llvm/ext/llvm-versions](https://github.com/crystal-lang/crystal/blob/master/src/llvm/ext/llvm-versions.txt) do not include 3.6.

I would have just opened a PR with improvements, but I haven't figured out the steps to fully install the compiler. I'm still stuck at some point in installing the compiler - see comment below which is more for troubleshooting than this documentation issue.

Note: I haven't worked with LLVM nor many open source toolchains/compilers/systems projects, forgive me if I've done anything dumb.

Environment
- OS: Ubuntu 20.04 focal(on the Windows Subsystem for Linux)
- Kernel: x86_64 Linux 4.19.104-microsoft-standard

My progress in figuring out how to install the Crystal compiler so far:
- ❌ `$ make clean crystal spec` 
  ```
  Makefile:58: *** Could not locate compatible llvm-config, make sure it is installed and in your PATH, or set LLVM_CONFIG. 
  Compatible versions: 11.1 11.0 10.0 9.0 8.0 7.1 6.0 5.0 4.0 3.9 3.8.  Stop.
  ```
  - ✔️ from https://apt.llvm.org/
  ```
  wget https://apt.llvm.org/llvm.sh
  chmod +x llvm.sh
  sudo ./llvm.sh 11
  ```
- ❌ `$ make clean crystal spec` (2nd time)
  ```
  ...
  Using compiled compiler at .build/crystal
  Using compiled compiler at .build/crystal
  
  /usr/bin/ld: cannot find -lxml2 (this usually means you need to install the development package for libxml2)
  /usr/bin/ld: cannot find -lyaml (this usually means you need to install the development package for libyaml)
  collect2: error: ld returned 1 exit status
  Error: execution of command failed with code: 1: `cc "${@}" -o /home/ ...
  ```
  - ✔️ `$ sudo apt-get install libxml2-dev libyaml-dev` 
- ❌ `$ make clean crystal spec` (3rd time)
  ```
  ...
  Using compiled compiler at .build/crystal
  Using compiled compiler at .build/crystal
  .build/all_spec 
  .build/all_spec: symbol lookup error: .build/all_spec: undefined symbol: _ZTVN10__cxxabiv117__class_type_infoE, version Spec::EqualExpectation(T)#failure_message<Tuple(String | Nil, Bool)>:String
  make: *** [Makefile:85: spec] Error 127
  ```
  - ✔️ `$ sudo apt install lld` from https://github.com/crystal-lang/crystal/issues/7177
- ❌ `$ make clean spec` (at this point I figured that it was `spec` causing issues and not the `compiler` phase)
  - ❌ Stuck here
  ```
    .build/all_spec 
  0.2.2
  ...CRYSTAL_PATH env variable does not contains /home/mysterise/projects/crystal/src
  ........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................*..........................................................................................................................................................................................................................................................................................................................................................................................................**....................................................................Invalid memory access (signal 11) at address 0x7fabaa4d99a6
  [0x561de29bbfe6] *Exception::CallStack::print_backtrace:(Int32 | Nil) +118
  [0x561de2083abc] __crystal_sigfault_handler +316
  [0x561de43ed2dc] sigfault_handler +44
  Segmentation fault
  make: *** [Makefile:85: spec] Error 139
  ```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -116,12 +116,15 @@ If you want to add/change something in the compiler,
 the first thing you will need to do is to [install the compiler](https://crystal-lang.org/install/).
 
 Once you have a compiler up and running, check that executing `crystal` on the command line prints its usage.
-Now you can setup your environment to compile Crystal itself, which is itself written in Crystal. Check out
-the `install` and `before_install` sections found in [.travis.yml](https://github.com/crystal-lang/crystal/blob/master/.travis.yml).
-These set-up LLVM 3.6 and its required libraries.
+Now you can setup your environment to compile Crystal itself, which is itself written in Crystal.
 
-Next, executing `make clean crystal spec` should compile a compiler and using that compiler compile and execute
-the specs. All specs should pass. You can use `make help` for a list of available make targets.
+The compiler needs [LLVM](https://llvm.org) and some other libraries. See [list of all required libraries](https://github.com/crystal-lang/crystal/wiki/All-required-libraries).
+
+Executing `make crystal` builds the compiler into `.build/compiler` and you can run it using the wrapper script at `bin/crystal`.
+The script sets up the proper environment variables that the compiler can find the standard library source files in `src/`.
+
+`make compiler_spec` runs the compiler specs. `make std_spec` runs the standard library specs.
+You can use `make help` for a list of available make targets.
 
 ## This guide
 

